x/tally/keeper/endblock.go

```diff
@@ -124,7 +127,13 @@ func (k Keeper) ProcessTallies(ctx sdk.Context, coreContract sdk.AccAddress) err
 			k.Logger(ctx).Debug("tally result", "request_id", req.ID, "tally_result", tallyResults[i])
 		}
 
-		processedReqs[req.ID] = distMsgs
+		baseFee := gasPrice.Mul(math.NewIntFromUint64(params.GasCostBase))
+		processedReqs[req.ID] = record.DistributionsWithBaseFee(baseFee)
+
+		ctx.EventManager().EmitEvent(sdk.NewEvent(types.EventTypeBaseFeeBurn,
+			sdk.NewAttribute(sdk.AttributeKeyAmount, baseFee.String()),
+		))
```

Comment:
It would be easier if all attributes get emitted under a single event. If that's too much of a headache than we should at least add the DR_ID as an attribute so we can link the event to the relevant DR.

Reply:
`DistributionsFromGasCalculation()` constructs a series of event attributes and emits them at once at the end. Since an attribute is just a key-value pair, proxy & executor gas attributes' values follow the "ID,amount" pattern.